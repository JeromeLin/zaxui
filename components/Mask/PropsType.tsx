interface BaseProps {
  visible?: boolean;
  type?: 'transparent' | 'light' | 'normal' | 'dark';
  onClose?: () => void;
  style?: React.CSSProperties;
}

export interface MaskProps extends BaseProps {
  prefixCls?: string;
  className?: string;
}
