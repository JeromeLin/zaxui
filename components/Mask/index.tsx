import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { IMask as MaskProps } from './PropsType';

export { MaskProps };

export default class Mask extends PureComponent<MaskProps, any> {
  static defaultProps = {
    prefixCls: 'za-mask',
    visible: false,
    type: 'normal',
  };

  render() {
    const { prefixCls, className, visible, type, onClose, ...others } = this.props;
    const markCls = classnames(`${prefixCls}`, className, type);

    return visible
      ? <div className={markCls} onClick={onClose} {...others} />
      : null;
  }
}
