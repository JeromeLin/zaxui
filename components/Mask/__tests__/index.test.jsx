import React from 'react';
import renderer from 'react-test-renderer';
import Mask from '../index';

describe('Mask', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Mask visible />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
