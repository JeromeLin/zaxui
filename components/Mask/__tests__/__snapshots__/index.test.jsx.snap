// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask renders correctly 1`] = `
<div
  className="za-mask normal"
  onClick={[Function]}
/>
`;
